From d66cea9e006191c062c7061ddfc549903bd6368a Mon Sep 17 00:00:00 2001
From: Michael Pauzner <mpauzner@habana.ai>
Date: Mon, 12 May 2025 14:50:44 +0300
Subject: [PATCH] [SW-221409] g3: arcmgmt: add lfs_image_in_ram

Add defconfig parameter for serdes LFS backup partition in RAM.

Change-Id: Ibc8fc4d03571ca5a38188e7cf3ed40b16831bb0a
Signed-off-by: Michael Pauzner <mpauzner@habana.ai>
---

diff --git a/boards/hl/gaudi3/hl_gaudi3_mgmt_asic_defconfig b/boards/hl/gaudi3/hl_gaudi3_mgmt_asic_defconfig
index 1d2470f..30d6738 100644
--- a/boards/hl/gaudi3/hl_gaudi3_mgmt_asic_defconfig
+++ b/boards/hl/gaudi3/hl_gaudi3_mgmt_asic_defconfig
@@ -156,6 +156,7 @@
 CONFIG_UART_CONSOLE_DEBUG_SERVER_HOOKS=y
 CONFIG_LFS_ENABLED=y
 CONFIG_LFS_OVER_SPI=y
+CONFIG_LFS_IMAGE_IN_RAM=y
 CONFIG_LFS_SHELL=y
 CONFIG_SHELL_HELP=y
 CONFIG_LFS_LOG_ENABLED_MANUALLY=n
diff --git a/modules/Kconfig.lfs b/modules/Kconfig.lfs
index 86037fb..dd0d219 100644
--- a/modules/Kconfig.lfs
+++ b/modules/Kconfig.lfs
@@ -8,6 +8,10 @@
 	bool "Enable LittleFS over SPI flash"
 	default n
 
+config LFS_IMAGE_IN_RAM
+	bool "Enable LittleFS in RAM"
+	default n
+
 config LFS_SHELL
 	bool "Enable LittleFS shell"
 	default n
